external/fv3fit/fv3fit/reservoir/model.py

```diff
@@ -98,6 +99,105 @@ def load(cls, path: str) -> "HybridReservoirComputingModel":
+        # TODO: centralize stacking logic for encoding decoding
+        # TODO: potentially use in train.py instead of special functions there
+        processed_inputs = self._input_data_to_array(inputs)  # x, y, feature dims
+        if self.model.rank_divider is None:
+            raise ValueError("Adapter currently require rank_divider to be set")
+
+        subdomains = self.model.rank_divider.flatten_subdomains_to_columns(
+            processed_inputs, with_overlap=True
+        )
+        flattened = flatten_2d_keeping_columns_contiguous(subdomains)
+        prediction = self.model.predict(flattened)
+        unstacked_arr = self.model.rank_divider.merge_subdomains(prediction)
+        return self._separate_output_variables(unstacked_arr)
+
+    # TODO: Add a synchronization facility and fix sync methods in models
+
+    def _encode_input_variables(self, inputs: xr.Dataset, autoencoder):
+        input_arrs = [
+            inputs[variable].values for variable in self.model.input_variables
+        ]
+
+        sample_dims_shape = list(input_arrs[0].shape[:-1])
+        feature_len = input_arrs[0].shape[-1]
+        stacked_sample_arrs = np.array(
+            [arr.reshape(-1, feature_len) for arr in input_arrs]
+        )
+        encoded = autoencoder.encode(stacked_sample_arrs)
+        encoded_shape = sample_dims_shape + [encoded.shape[-1]]
+        return encoded.reshape(encoded_shape)
+
+    def _join_input_variables(self, inputs: xr.Dataset):
+        input_arrs = [inputs[variable] for variable in self.model.input_variables]
+        joined_feature_inputs = np.concatenate(input_arrs, axis=-1)
+
+        return joined_feature_inputs
+
+    def _input_data_to_array(self, inputs: xr.Dataset):
+        if self._input_feature_sizes is None:
+            self._input_feature_sizes = [
+                inputs[key].shape[-1] for key in self.model.input_variables
+            ]
+
+        if self.model.autoencoder is not None:
+            arr = self._encode_input_variables(inputs, self.model.autoencoder)
+        else:
+            arr = self._join_input_variables(inputs)
+
+        return arr
+
+    def _separate_output_variables(self, outputs: np.ndarray):
+        if self.model.autoencoder is not None:
+            ds = self._decode_ouput_variables(outputs, self.model.autoencoder)
+        else:
+            ds = self._separate_output_from_stacked_array(outputs)
+
+        return ds
+
+    def _decode_ouput_variables(self, encoded_output: np.ndarray, autoencoder):
+        if encoded_output.ndim > 3:
+            raise ValueError("Unexpected dimension size in decoding operation.")
+
+        feature_size = encoded_output.shape[-1]
+        encoded_output = encoded_output.reshape(-1, feature_size)
+        decoded_arrs = autoencoder.decode(encoded_output)
+
+        return self._separate_output_from_stacked_array(decoded_arrs)
+
+    def _separate_output_from_stacked_array(self, outputs: np.ndarray):
```

Comment:
I think currently the dense autoencoder transformers behave as desired (return separate arrays for each output) whereas the sklearn transformer does not. Since we decided to fix this in the near future to have all `Transformer` objects have the former behavior, I'm ok with omitting this functionality from this PR (i.e. assume that the output is already in separate arrays, and it's ok if the sklearn transform does not work with this adapter in the meantime) so that it doesn't have to be altered again in the near future when the Transformers are all made consistent.